py_modules/imap_api.py

```diff
@@ -535,3 +534,33 @@ def search_mail(self, text):
                 return True, data
             else:
                 return False, []
+
+    def mail_watcher(self):
+        mailserver = imaplib.IMAP4_SSL(self.server)
+        mailserver.login(self.userName, self.password)
+        result, message = mailserver.select(readonly=True)
+        if result != 'OK':
+            raise Exception(message)
+        retcode, data = mailserver.search(None, '(UNSEEN)')
+        data[0] = data[0].decode("utf-8")
+        if data[0]:
+            last_uid = str(str.split(str(data[0]))[-1])
+            logger.debug(f'Latest received email uid: {last_uid}')
+            last_local_uid = DBApi().get_last_email_id(1)[0][0]
```

Comment:
Hello! The function called fails if the database doesn't contains mails.

`  File "/home/simone/PycharmProjects/ToooMail/py_modules/imap_api.py", line 551, in mail_watcher
    last_local_uid = DBApi().get_last_email_id(1)[0][0]`

Reply:
Solved: 34fb55271ad2b06ad0f28edf1fbb134ced6a361b
Linked Issue: #103